common/source/docs/common-blheli32-passthru.rst

```diff
@@ -1,62 +1,158 @@
+
+  - :ref:`SERVO_BLH_OTYPE<SERVO_BLH_OTYPE>` : This needs to be set to the protocol being used for the DShot protocol being used on those additional outputs if not the same as the normal copter style motor outputs.
+
+- :ref:`SERVO_BLH_TRATE <SERVO_BLH_TRATE>` defaults to 10 and normally does not need to be changed. this enables telemetry at a 10Hz update rate from the ESC.
+
+- :ref:`SERVO_BLH_POLES <SERVO_BLH_POLES>` defaults to 14 which applies to the majority of brushless motors and normally does not need to be changed.  Adjust as required if you're using motors with a pole count other than 14 to calculate true motor shaft RPM from ESC's e-field RPM.
+
+ESC Telemtry Logging and Reporting
+----------------------------------
+
+The autopilot requests status information from one ESC at a time, cycling between them. This information is logged to the onboard log's ESCn messages and can be viewed in any :ref:`ArduPilot compatible log viewer <common-logs>`.
+
+- RPM
+- Voltage
+- Current
+- Temperature
+- Total Current
+
+The RCOU messages are also written to the onboard logs which hold the requested output level sent to the ESCs expressed as a number from 1000 (meaning stopped) to 2000 (meaning full output).
+
+This data can also be viewed in real-time using a ground station.  If using the Mission Planner go to the Flight Data screen's status tab and look for esc1_rpm.
+
+.. image:: ../../../images/dshot-realtime-esc-telem-in-mp.jpg
+    :target: ../_images/dshot-realtime-esc-telem-in-mp.jpg
+    :width: 450px
+
+.. note::
+
+   Sending BLHeli32 telemetry data to the GCS requires the telemetry connection use MAVLink2.  ArduPilot uses MAVLink2 by default on the USB port but if another port is used it may be necessary to set the SERIALx_PROTOCOL parameter to 2 (where "x" is the serial port number used for the telemetry connection).
+
+In addition, some telemetry values can be displayed on the integrated :ref:`on-board OSD <common-osd-overview>`, if your autopilot has one.
+
+.. _esc-telemetry-based-battery-monitor:
+
+Use as Battery Monitor
+======================
+
+By setting a battery monitor instance to BLHeli32 ESC  type (for example :ref:`BATT2_MONITOR<BATT2_MONITOR>` = 9), all connected BLHeli32 ESCs with connected telemetry wiring to the configured autopilot serial port, will be aggregated as a single source. The voltages reported will be averaged, the currents totaled, and the consumed current accumulated.
+
+.. _bidir-dshot:
+
+Bi-directional Dshot
+====================
+
+Newer versions of BLHeli32 (32.7 and higher) and BLHeli_S (16.73 and higher) support returning motor RPM values over the Dshot signal line. Supporting bi-directional Dshot requires exclusive use of one or more DMA channels and thus not all versions of ArduPilot support it. Versions that support bi-directional Dshot natively are `BeastH7`, `BeastF7` and `KakuteF7Mini`, other firmware versions end in "-bdshot" to indicate support for bi-directional Dshot.
```

Comment:
Also KakuteF4Mini, KakuteH7Mini, BeastH7v2, BeastF7v2, FlywooF745, FlywooF745Nano

Reply:
cool, I've added these to the list and slightly changed the wording to make it more clear that for other autopilots a special version of ArduPilot needs to be loaded on the autopilot.